addons/account_qr_code_emv/models/res_bank.py

```diff
@@ -39,24 +48,25 @@ def _get_qr_vals(self, qr_method, amount, currency, debtor_partner, free_communi
             additional_data_field = self._get_additional_data_field(comment)
 
             qr_code_vals = [
-                '000201',                                                                           # Payload Format Indicator
-                '010212',                                                                           # Dynamic QR Codes
-                f'26{len(merchant_account_info):02}{merchant_account_info}',                        # Merchant Account Information
-                '52040000',                                                                         # Merchant Category Code
-                f'5303{currency_code}',                                                             # Transaction Currency
-                f'54{len(str(amount)):02}{amount}',                                                 # Transaction Amount
-                f'5802{self.partner_id.country_code}',                                              # Country Code
-                f'59{len(merchant_name):02}{merchant_name}',                                        # Merchant Name
-                f'60{len(merchant_city):02}{merchant_city}',                                        # Merchant City
+                (0, '01'),                                                          # Payload Format Indicator
+                (1, '12'),                                                          # Dynamic QR Codes
+                (26, merchant_account_info),                                        # Merchant Account Information
+                (52, '0000'),                                                       # Merchant Category Code
+                (53, currency_code),                                                # Transaction Currency
+                (54, amount),                                                       # Transaction Amount
+                (58, self.partner_id.country_code),                                 # Country Code
+                (59, merchant_name),                                                # Merchant Name
+                (60, merchant_city),                                                # Merchant City
+                (62, additional_data_field),                                        # Additional Data Field
             ]
 
-            if additional_data_field:
-                qr_code_vals.append(f'62{len(additional_data_field):02}{additional_data_field}')    # Additional Data Field
+            qr_code_str = ''.join([self._serialize(*val) for val in qr_code_vals])
+            qr_code_str += '6304'                                                   # CRC16
+
+            crc = self._get_crc16(bytes(qr_code_str, 'utf-8'))
```

Comment:
```suggestion
            qr_code_str += '6304'                                                   # CRC16
            crc = self._get_crc16(bytes(qr_code_str, 'utf-8'))
```
The comment at the end of the line + the space makes it confusing, it seems like it's two different things